aws/flatcar-linux/kubernetes/cl/controller.yaml.tmpl

```diff
@@ -156,6 +157,55 @@ storage:
             --net=host \
             --dns=host \
             --exec=/bootkube -- start --asset-dir=/assets "$@"
+    - path: /opt/etcd-rejoin
+      filesystem: root
+      mode: 0555
+      contents:
+        inline: |
+          #!/bin/bash
+          set -eou pipefail
+          # Rejoin a cluster as fresh node when etcd cannot join
+          # (e.g., after repovisioning, crashing or node being down).
+          # Set ExecStopPost=-/opt/etcd-rejoin to run when etcd failed and
+          # use env vars of etcd-member.service.
+          # Skip if not provisioned
+          if [ ! -d "/etc/ssl/etcd/" ]; then exit 0; fi
+          # or got stopped.
+          if [ "$EXIT_CODE" = "killed" ]; then exit 0; fi
+          now=$(date +%s)
+          if [ -f /var/lib/etcd-last-fail ]; then
+            last=$(cat /var/lib/etcd-last-fail)
+          else
+            last=0
+          fi
+          echo "$now" > /var/lib/etcd-last-fail
+          let "d = $now - $last"
+          # Skip and restart regularly if it does not fail within 120s.
```

Comment:
Can't we tune the retry times on the systemd unit instead of here? I thought there was some functionality for that, but not sure

Reply:
Yes, but that's a different thing. They restart the service after waiting for 10s. But this is to detect how long ago the last failure was. If the service failed twice in two minutes, it's rejoining as fresh node, otherwise it just restarts regularly.

Reply:
Oh, I thought this was only to wait 2 minutes. I keep forgetting this is the script that gets executed after :man_facepalming: